pkgs/development/compilers/llvm/common/default.nix

```diff
@@ -400,8 +400,21 @@ let
       };
     }
     // lib.optionalAttrs (lib.versionAtLeast metadata.release_version "20") {
-      flang = callPackage ./flang {
+      flang-unwrapped = callPackage ./flang {
         mlir = tools.mlir;
+        # The C++ template instantiations in flang frequently cause g++ to be
+        # OOM-killed, so use clang++ instead.
```

Comment:
hmm, on a machine with how much memory?

Reply:
This comment was originally from @picostove, I'm not sure about the exact memory specs. I've encountered similar g++ OOM issues with olberg bot in a past flang-related PR though.
But I think it's better to remove this override as of now.